src/btouch.cs

```diff
@@ -376,16 +376,25 @@ static int Main2 (string [] args)
 			CurrentPlatform = PlatformName.MacOSX;
 			Unified = true;
 			nostdlib = true;
-			skipSystemDrawing = target_framework == TargetFramework.Xamarin_Mac_4_5_Full;
 			if (string.IsNullOrEmpty (baselibdll)) {
-				if (target_framework == TargetFramework.Xamarin_Mac_2_0_Mobile) {
+				if (target_framework == TargetFramework.Xamarin_Mac_2_0_Mobile)
 					baselibdll = Path.Combine (GetSDKRoot (), "lib", "reference", "mobile", "Xamarin.Mac.dll");
-					AddAnyMissingSystemReferencesFromSDK ("lib/mono/Xamarin.Mac", references);
-				} else {
+				else if (target_framework == TargetFramework.Xamarin_Mac_4_5_Full)
+					baselibdll = Path.Combine (GetSDKRoot (), "lib", "reference", "full", "Xamarin.Mac.dll");
+				else if (target_framework == TargetFramework.Xamarin_Mac_4_5_System)
```

Comment:
Is the `else` clause possible here?

In any case we should do something about it (either throw an exception, or merge the previous two conditions into a single `else`, since the nested block is identical in both cases).

I've found that `else` clauses which aren't implemented because they're not possible have a tendency to bite hard one day :smile:

Reply:
Both of them are inside `case "xamarin.mac":` so it'd have to be a new target framework.

A throw is a good idea. The blocks are slightly different however.

Reply:
They both do this:

```csharp
baselibdll = Path.Combine (GetSDKRoot (), "lib", "reference", "full", "Xamarin.Mac.dll");
```

was it intentional?

Also we already have a proper error for this condition, which we use elsewhere: https://github.com/xamarin/xamarin-macios/blob/76f0153e720d04dcd8ecc54d12bdba69d915ff75/src/btouch.cs#L108